runtime/near-vm-logic/src/logic.rs

```diff
@@ -628,6 +630,28 @@ impl<'a> VMLogic<'a> {
         Ok(self.context.block_index)
     }
 
+    /// Returns the current block hash.
+    ///
+    /// # Cost
+    ///
+    /// `base`
+    #[cfg(feature = "protocol_feature_vm_hash")]
+    pub fn block_hash(&mut self) -> Result<BlockHash> {
+        self.gas_counter.pay_base(base)?;
+        Ok(self.context.block_hash)
+    }
+
+    /// Returns the previous block hash.
+    ///
+    /// # Cost
+    ///
+    /// `base`
+    #[cfg(feature = "protocol_feature_vm_hash")]
+    pub fn prev_block_hash(&mut self) -> Result<BlockHash> {
```

Comment:
During discussion with nearmax, we realized that returning a `BlockHash` directly might not *just* work. All our other APIs only return primitive types (u64 integers), that's the first case where we want to return something bigger. Looks like we need either to use an out-parameter here, or [multiple return values](https://github.com/WebAssembly/multi-value), the latter approach being more promising imo.

Reply:
The pattern used in other examples, [such as `sha256`](https://github.com/near/nearcore/blob/d336b3fc1b9dc2167aa2eb9a89e24e4a2a09e27d/runtime/near-vm-logic/src/logic.rs#L901), is to use the register API, so I think we should probably stick to that for consistency. Though wasm handling multiple return values now is really cool!

Reply:
Ah, I somehow miss the fact that we already do use out parameters. Clearly, we should use the same design here as well.